### ISSUE TITLE
Access and restore archive metadata

### ISSUE THREAD

**Issue by tlaurion** [tasket/sparsebak#38]
Reinstalled Qubes 4.0.2RC1 from scratch. Previously done a native QubesOS backup to restore needed VMs. Redeployed sparsebak. 

Was expecting to be able to use sparsebak arch-init that would re-add all archived volumes present in backup subdir and be able to call send receive and prune. It doesn't seem so. 

Couple of comments:
- Why isn't there a way to restore configuration from configured archive? I would have expected to be able to sync archive.ini back and automatically add volumes for all volumes present in the backup archive directory, even the ones not being present on host to proceed with the restoration of volumes when desired (qvm-create first, then receive to /dev/qubes_dom0/vm-volume)
- After having added back all volumes manually, the existing volumes present on both archive and host are not updating the backup, but creating a new one.
- After having send all new backups, doing a "prune --all-before session=*now*" doesn't prune old backups at all, consuming space I currently don't know if it is safe to manually delete (eg: vm-whonix-gw-15-root that got replaced with new template)
- It would be lovely to know what is stored under /var/lib/sparsebak and why it is not stored under backup subdir. Logically to me, everything should be stored in backup subdir to permit easy restoration.
- QubesOS xml parsing should be added, so that a vmname can be specified to be backed up and restored on a fresh install.

Let me know if I need to be clearer :)

**Comment by tasket** [comment]
@tlaurion The sparsebak alpha doesn't have a 'restore metadata from archive' function.

However, you can copy the archive metadata back into the system:

1. Remove the new metadata dir with `rm -r /var/lib/sparsebak`

2. Also delete from the archive any new session dirs you made since you reinstalled the system.

3. Look at the 'volinfo' files for the volumes you tried to back up after reinstalling. The `first` and `last` lines should be changed to reflect the earliest and latest 'S_*' session dirs for the volume. –– At this point, the archive should be back to its 'original' state before the system reinstallation.

4. Then copy the archive 'sparsebak' dir hierarchy _minus_ the data chunks to /var/lib/sparsebak. You can do this with `tar` like so:
```
tar --exclude ?????????/x* --exclude *.tar -cvf sb_metadata.tar /path/to/sparsebak
cd /var/lib
sudo tar -xf sb_metadata.tar
```

Once the metadata is restored to /var/lib/sparsebak you can perform normal `receive` operations. In order for sparsebak to re-create the snapshot and delta file for a volume (so you can resume using `send` as usual) you need to restore the volume to the original path like so:
```
sparsebak.py --save-to=/dev/qubes_dom0/vm-work-private receive vm-work-private
```

It should say "Initial snapshot created for vm-work-private" when finished. Unfortunately, there is no way (yet) to tell sparsebak to use the original volume path, but I plan to add that soon along with a way to restore archive metadata.

**Comment by tasket** [comment]
In case you're wondering why this capability isn't in sparsebak alpha, I did add a similar ability to the beta1 candidate. This is the renamed program, now called **`wyng`**. It can restore volumes from archives that are not currently configured in the system, but this just gets you the volume and not the config+metadata. Adding metadata restoration will be the next step.

**Comment by tasket** [comment]
Implemented with commit 8fd222f2b8e3759a353c08abfdcc26b2b4e4e82a
